src/magic_folder/participants.py

```diff
@@ -149,15 +154,34 @@ def _is_self(self, dirobj):
 @implementer(IParticipant)
 @attr.s(frozen=True)
 class _CollectiveDirnodeParticipant(object):
+    """
+    An ``IParticipant`` implementation backed by a Tahoe-LAFS directory node
+    (a DMD).
+
+    :ivar unicode name: A human-readable identifier for this participant.  It
+        will be the name of the DMD directory in the collective.
+
+    :ivar allmydata.interfaces.IDirectoryNode dirobj: An object for accessing
+        the Tahoe-LAFS directory node containing this participant's files.
+
+    :ivar bool is_self: True if this participant is known to represent the
+        ourself, False otherwise.  Concretely, "ourself" is whoever can write
+        to the directory node.
+    """
     name = attr.ib(validator=attr.validators.instance_of(unicode))
     dirobj = attr.ib()
```

Comment:
This could benefit from a validator (there is already support for zope-interfaces right?)